docs/source/examples/learning-depolarizing-noise.myst

```diff
@@ -0,0 +1,210 @@
+  name: python3
+---
+
+# Learning quasiprobability representations with a depolarizing noise model
+
+In this example, we demonstrate the workflow of learning quasiprobability representations of a `CNOT` gate with a depolarizing noise model,
+from Clifford circuit data.
+The depolarizing noise model is parameterized by the noise strength, `epsilon`.
+The resulting quasiprobability representations of the `CNOT` gate are then used to obtain an error-mitigated expectation value with Mitiq's
+{ref}`probabilistic error cancellation module <guide/pec/pec>`. 
+For a more in-depth description of the learning function used in this example, see the section on {func}`.learn_depolarizing_noise_parameter`
+in the API-doc.
+The learning-based PEC workflow was inspired by the procedure described in *Strikis et al. PRX Quantum (2021)* {cite}`Strikis_2021_PRXQuantum`.
+
+```{code-cell} ipython3
+import numpy as np
+from matplotlib import pyplot as plt
+from cirq import (
+    CXPowGate,
+    DepolarizingChannel,
+    LineQubit,
+    Circuit,
+)
+from mitiq import Executor, Observable, PauliString
+from mitiq.interface.mitiq_cirq import compute_density_matrix
+from mitiq.cdr import generate_training_circuits
+from mitiq.cdr._testing import random_x_z_cnot_circuit
+from mitiq.pec import execute_with_pec
+from mitiq.pec.representations.depolarizing import (
+    represent_operation_with_local_depolarizing_noise,
+)
+from mitiq.pec.representations.learning import (
+    biased_noise_loss_function,
+    learn_depolarizing_noise_parameter,
+)
+```
+
+Since the learning-based workflow uses the function {func}`.cdr.generate_training_circuits` from Mitiq's {ref}`Clifford data regression module
+<guide/cdr/cdr>` to generate the near-Clifford training circuits, the input circuit must be one that is compiled into the Rx-Rz-CNOT gateset. 
+Here we use a simple Rx-Rz-CNOT circuit, with an (optional) seed for reproducibility.
+
+```{code-cell} ipython3
+circuit = random_x_z_cnot_circuit(
+    LineQubit.range(2), n_moments=10, random_state=np.random.RandomState(1)
+)
+print(circuit)
+```
+
+Define the ideal executor function for simulating the Clifford training circuits without noise. 
+This will be used for comparison to the error-mitigated expectation values in the learning function, and for a final comparison with the
+mitigated and unmitigated values at the end of the workflow.
+
+```{code-cell} ipython3
+def ideal_execute(circ: Circuit) -> np.ndarray:
+    # For large circuits, this should be replaced by a near-Clifford simulator.
+    return compute_density_matrix(circ, noise_level=(0.0,))
+
+
+ideal_executor = Executor(ideal_execute)
+```
+
+Define the noisy executor, in this case for simulating depolarizing noise on the `CNOT` gate in each of the training circuits. 
+The optimized value of the noise strength `epsilon` should be close to the value defined for this executor.
+
+```{code-cell} ipython3
+CNOT_ops = list(circuit.findall_operations_with_gate_type(CXPowGate))
```

Comment:
Casting the generator to list is now unneeded.

Reply:
Thanks for asking about this! I will leave it as-is because if I don't cast to list, only one noisy operation is added in the `noisy_executor`.

Reply:
Fine to leave it out, but are you sure? You later iterate over `CNOT_ops`, rather than just grabbing the first element. [`findall_operations_with_gate_type`](https://github.com/quantumlib/Cirq/blob/946f247dc66f1257b463766536a854bb36b4c3c6/cirq-core/cirq/circuits/circuit.py#L778-L780) returns an iterable so I think it should be fine to leave it as that.

Reply:
I thought so too, but when I tested it, the loss function plot shifted so that the minimum was around 0.03 instead of the true value of 0.05.  I think that's because only one noisy op was added in the executor if I left CNOT_ops as an iterable.

Reply:
Ohhh okay I see. Yeah we can't iterate over a generator multiple times. Forgot about that! Thanks for helping me get there.